llvm/lib/Target/RISCV/RISCVVLOptimizer.cpp

```diff
@@ -31,6 +31,55 @@ using namespace llvm;
 
 namespace {
 
+struct VLInfo {
+  std::variant<Register, int64_t> VL;
+
+  VLInfo(const MachineOperand &VLOp) {
+    if (VLOp.isImm())
+      VL = VLOp.getImm();
```

Comment:
I think a Register can be constructed via an int64_t, is std::variant definitely storing the right type here?